src/future/timeout.rs

```diff
@@ -0,0 +1,78 @@
+use std::error::Error;
+use std::fmt;
+use std::pin::Pin;
+use std::time::Duration;
+
+use futures_timer::Delay;
+use pin_utils::unsafe_pinned;
+
+use crate::future::Future;
+use crate::task::{Context, Poll};
+
+/// Awaits a future or times out after a duration of time.
+///
+/// # Examples
+///
+/// ```no_run
+/// # #![feature(async_await)]
+/// # fn main() -> std::io::Result<()> { async_std::task::block_on(async {
+/// #
+/// use std::time::Duration;
+///
+/// use async_std::future;
+///
+/// let never = future::pending::<()>();
+/// let dur = Duration::from_secs(5);
```

Comment:
```suggestion
/// let dur = Duration::from_millis(5);
```

If we make the duration real short we can actually run this assertion! (: